packages/docusaurus/bin/docusaurus.mjs

```diff
@@ -166,33 +146,14 @@ cli
   .option('-p, --port <port>', 'use specified port (default: 3000)')
   .option('--build', 'build website before serving (default: false)')
   .option('-h, --host <host>', 'use specified host (default: localhost)')
-  .action(
-    async (
-      siteDir,
-      {
-        dir = 'build',
-        port = 3000,
```

Comment:
where are these default values being applied now?

isn't it convenient that the default value and the doc message mentioning it are co-located?


Why not doing the opposite instead: always apply all default values in this file, ensuring we don't have to handle partial options anywhere else in the system?

Reply:
These have always been applied internally, before and after. These default values actually only serve the purpose of being confusing & overly defensive.

Reply:
as far as I understand this is the only place where `3000` is defined. 

Otherwise it's the default of the underlying lib, which as you can see, can change over time (https://github.com/vercel/serve/pull/680)

I'd rather keep applying these defaults explicitly so that we don't depend on underlying deps changes.

Also we can use the default options in commander help messages

```
  .option('-p, --port <port>', `use specified port (default: ${DefaultServeOptions.port})`)
```

Reply:
Oh... you're sure? 😄

https://github.com/facebook/docusaurus/blob/3b32a41f229a1cbeb869ba4232035af3ad6b8a37/packages/docusaurus/src/commands/commandUtils.ts#L22

Reply:
ah 😅  didn't search in utils